common/web/gesture-recognizer/src/tools/recorder/build.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+#
+# Builds the include script for the current Keyman version.
+#
+
+# Exit on command failure and when using unset variables:
+set -eu
+
+## START STANDARD BUILD SCRIPT INCLUDE
+# adjust relative paths as necessary
+THIS_SCRIPT="$(greadlink -f "${BASH_SOURCE[0]}" 2>/dev/null || readlink -f "${BASH_SOURCE[0]}")"
+. "$(dirname "$THIS_SCRIPT")/../../../../../../resources/build/build-utils.sh"
+## END STANDARD BUILD SCRIPT INCLUDE
+
+. "$KEYMAN_ROOT/resources/shellHelperFunctions.sh"
+
+# This script runs from its own folder
+cd "$(dirname $THIS_SCRIPT)"
+
+display_usage ( ) {
+  echo "Usage: $0 [configure] [clean] [build] [test]"
+  echo "          [--verbose|-v]"
+  echo "       $0 -h|--help"
+  echo
+  echo "  clean                  removes build/ folder"
+  echo "  build                  builds wrapped version of package"
+  echo "                           [if required will: configure]"
+}
+
+build_recorder_page ( ) {
+  cp src/pageStyle.css    build/pageStyle.css
+  cp src/recorder.js      build/recorder.js
+
+  cp ../host-fixture/gestureHost.css  build/gestureHost.css
+
+  # Thanks to https://stackoverflow.com/a/10107668 for this tidbit.
+  # Searches for FIXTURE_TARGET above and replaces it with the actual fixture!
+  node update-index.js build/index.html
```

Comment:
I don't understand why we need all the complexity of update-index.js and host-fixture for this? Why do we need to construct all this each time -- it seems like a lot of code to maintain.

Reply:
1.  To perform in-browser unit-tests, especially ones that match the user tests defined in one of this PR's ancestors, we _need_ an HTML fixture.  That much is pretty unavoidable.

2.  To keep our maintenance load _down_, it's best if that fixture match what is used on the recorder page.  The unit tests may then function by simply "playing back" the recording on the same element configuration and verifying that we get the same results.

3.  For maintaining that fixture... note that the fixture _will not function properly_ without its associated CSS.  It's difficult to maintain those elements in isolation when the CSS isn't linked.  (Of course, having a WET copy with it linked, then doing the ol' CTRL-C CTRL-V works well enough.)  Took me a while to get the element configuration + CSS structure functional and to its current state, after all.

4. Since our team policy is generally "write DRY code when possible," this turned out to be the least complex / most maintainable way I could discern to meet that goal.  We clearly need the same 'fixture' for both unit tests and unit-test resource recording.  (It's also a near-duplicate with the user-testing 'host page, technically.)  I could easily see updating the recorder page but forgetting to update the fixture at some point in the future; why not prevent that?

As far as the "maintenance" angle goes...

At one stage in development, I was using `sed` and `awk` to extract and insert, respectively.  That said, my `sed` and `awk` skills are quite low, _and_ the scripts broke instantly on Mac due to differences with the versions it natively includes.

On the flip-side, we're already using Node, JS, and TS, and we have code using the modules and functions in that script already.  Also, I personally find it much easier to parse at a glance than the dense `sed` and `awk` constructions I "Frakensteined" together via StackOverflow references.

Now, one idea that may be slightly more palatable - what if the fixture were extracted _from_ the recorder page, rather than defined in its own separate subfolder?  We'd still need the extractor, to be sure, and it's worth noting that this proposal would technically overload the purpose of the tools/recorder subfolder.